### PR TITLE
add wasm extension

### PR DESCRIPTION
**Pull request by sohankunkerkar**
xref: https://issues.redhat.com/browse/RFE-4177
related to https://github.com/openshift/os/pull/1327

**- What I did**
This PR adds `crun-wasm` to the list of extensions.

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: pkg/daemon/update.go [hunk 1]
```diff
@@ -1073,6 +1073,7 @@ func getSupportedExtensions() map[string][]string {
 	// These are RHCOS supported extensions.
 	// Each extension keeps a list of packages required to get enabled on host.
 	return map[string][]string{
+		"crun-wasm":            {"crun-wasm"},
```
Comment (haircommander) [inline pkg/daemon/update.go:1076]: I think it's worth thinking about this name. For instance, `sandboxed-containers` is name for installing `kata`. should we name this `wasm` or `wasm-container-runtime` or something?

Comment (sohankunkerkar) [reply pkg/daemon/update.go]: Yeah, I will stick to `wasm`.

File: test/e2e-single-node/sno_mcd_test.go [hunk 2]
```diff
@@ -196,7 +196,7 @@ func TestExtensions(t *testing.T) {
 			Config: runtime.RawExtension{
 				Raw: helpers.MarshalOrDie(ctrlcommon.NewIgnConfig()),
 			},
-			Extensions: []string{"ipsec", "usbguard", "kernel-devel", "kerberos"},
+			Extensions: []string{"crun-wasm", "ipsec", "usbguard", "kernel-devel", "kerberos"},
```
Comment (haircommander) [inline test/e2e-single-node/sno_mcd_test.go:199]: isn't this `wasm`?

**Comment by sohankunkerkar** [comment]
@haircommander could you take another look?

**Review by yuqi-zhang — APPROVED** [review]

**Pull request merged**